tests/bluetooth/bsim_bt/bsim_test_host/src/main_eatt.c

```diff
@@ -0,0 +1,295 @@
+	if (err) {
+		FAIL("Can't enable Bluetooth (err %d)\n", err);
+	}
+
+	err = bt_le_adv_start(BT_LE_ADV_CONN_NAME, ad, ARRAY_SIZE(ad), NULL, 0);
+	if (err) {
+		FAIL("Advertising failed to start (err %d)\n", err);
+	}
+
+	while (!is_connected) {
+		k_sleep(K_MSEC(100));
+	}
+
+	struct bt_att_cb att_cb = {
+		.att_chan_connected = att_chan_connected,
+		.att_chan_disconnected = att_chan_disconnected,
+	};
+
+	bt_att_cb_register(&att_cb);
+
+	while (!security_done) {
+		k_sleep(K_MSEC(100));
+	}
+
+	while (num_eatt_channels < CONFIG_BT_EATT_MAX) {
+		k_sleep(K_MSEC(100));
+	}
+
+	/* Disconnect */
+	err = bt_conn_disconnect(default_conn, BT_HCI_ERR_REMOTE_USER_TERM_CONN);
+	if (err) {
+		FAIL("Disconnection failed (err %d)\n", err);
+	}
+
+	while (is_connected) {
+		k_sleep(K_MSEC(100));
+	}
+
+	if (num_eatt_channels) {
+		FAIL("EATT channels still connected (%d)\n", num_eatt_channels);
+	}
+
+	PASS("EATT Peripheral tests Passed\n");
+}
+
+static void device_found(const bt_addr_le_t *addr, int8_t rssi, uint8_t type,
+			 struct net_buf_simple *ad)
+{
+	struct bt_le_conn_param *param;
+	int err;
+
+	err = bt_le_scan_stop();
+	if (err) {
+		FAIL("Stop LE scan failed (err %d)\n", err);
+	}
+
+	param = BT_LE_CONN_PARAM_DEFAULT;
+	err = bt_conn_le_create(addr, BT_CONN_LE_CREATE_CONN, param, &default_conn);
+
+	if (err) {
+		FAIL("Create conn failed (err %d)\n", err);
+	}
+	printk("Device connected\n");
+}
+
+static void test_central_main(void)
+{
+	struct bt_le_scan_param scan_param = {
+		.type = BT_LE_SCAN_TYPE_ACTIVE,
+		.options = BT_LE_SCAN_OPT_NONE,
+		.interval = BT_GAP_SCAN_FAST_INTERVAL,
+		.window = BT_GAP_SCAN_FAST_WINDOW,
+	};
+
+	int err;
+
+	err = bt_enable(NULL);
+	if (err) {
+		FAIL("Can't enable Bluetooth (err %d)\n", err);
+	}
+
+	err = bt_le_scan_start(&scan_param, device_found);
+	if (err) {
+		FAIL("Scanning failed to start (err %d)\n", err);
+	}
+
+	while (!is_connected) {
+		k_sleep(K_MSEC(100));
+	}
+
+	struct bt_att_cb att_cb = {
+		.att_chan_connected = att_chan_connected,
+		.att_chan_disconnected = att_chan_disconnected,
+	};
+
+	bt_att_cb_register(&att_cb);
+
+	err = bt_conn_set_security(default_conn, CONFIG_BT_EATT_SEC_LEVEL);
+	if (err) {
+		FAIL("Security change failed");
+	}
+
+	while (!security_done) {
+		k_sleep(K_MSEC(100));
+	}
+
+	err = bt_eatt_connect(default_conn, CONFIG_BT_EATT_MAX);
+	if (err < 0) {
+		FAIL("bt_eatt_connect failed (err: %d)\n", err);
+	}
+
+	while (num_eatt_channels < CONFIG_BT_EATT_MAX) {
+		k_sleep(K_MSEC(100));
+	}
+
+	/* Wait for disconnect */
+	while (is_connected) {
+		k_sleep(K_MSEC(100));
+	}
+
+	if (num_eatt_channels) {
+		FAIL("EATT channels still connected (%d)\n", num_eatt_channels);
+	}
+
+	PASS("EATT Central tests Passed\n");
+}
+
+static void test_init(void)
+{
+	bst_ticker_set_next_tick_absolute(60e6); /* 60 seconds */
+	bst_result = In_progress;
+}
+
+static void test_tick(bs_time_t HW_device_time)
+{
+	if (bst_result != Passed) {
+		FAIL("Too few EATT channels connected\n");
```

Comment:
Is this failure reason correct? If bst_result != Passed, then it fails due to two few EATT channels connected?